src/com/com_nrf24l01_core.cpp

```diff
@@ -0,0 +1,312 @@
+  auto get_config_register = spi_protocol_->ReadRegister(reg::config::REG_ADDR);
+  if (get_config_register.first != types::DriverStatus::OK) {
+    return get_config_register.first;
+  }
+  utilities::Byte config_register(get_config_register.second);
+
+  if (mode == OperationMode::PRIM_TX) {
+    config_register.ClearBit(reg::config::PRIM_RX);
+    current_operation_mode_ = OperationMode::PRIM_TX;
+  } else {
+    config_register.SetBit(reg::config::PRIM_RX);
+    current_operation_mode_ = OperationMode::PRIM_RX;
+  }
+
+  return spi_protocol_->WriteRegister(reg::config::REG_ADDR, config_register.Get());
+}
+
+auto NRF24L01Core::SetRFChannel(std::uint8_t channel) noexcept -> types::DriverStatus {
+  return spi_protocol_->WriteRegister(reg::rf_ch::REG_ADDR, channel);
+}
+
+auto NRF24L01Core::SetDataRate(DataRateSetting data_rate) noexcept -> types::DriverStatus {
+  auto get_config_register = spi_protocol_->ReadRegister(reg::rf_setup::REG_ADDR);
+  if (get_config_register.first != types::DriverStatus::OK) {
+    return get_config_register.first;
+  }
+  utilities::Byte config_register(get_config_register.second);
+
+  if (data_rate == DataRateSetting::RF_DR_1MBPS) {
+    config_register.ClearBit(reg::rf_setup::RF_DR);
+  }
+  if (data_rate == DataRateSetting::RF_DR_2MBPS) {
+    config_register.SetBit(reg::rf_setup::RF_DR);
+  }
+
+  return spi_protocol_->WriteRegister(reg::rf_setup::REG_ADDR, config_register.Get());
+}
+
+auto NRF24L01Core::EnableCRC() noexcept -> types::DriverStatus {
+  auto get_config_register = spi_protocol_->ReadRegister(reg::config::REG_ADDR);
+  if (get_config_register.first != types::DriverStatus::OK) {
+    return get_config_register.first;
+  }
+  utilities::Byte config_register(get_config_register.second);
+  config_register.SetBit(reg::config::EN_CRC);
+
+  return spi_protocol_->WriteRegister(reg::config::REG_ADDR, config_register.Get());
+}
+
+auto NRF24L01Core::SetCRCEncodingScheme(CRCEncodingScheme encoding_scheme) noexcept -> types::DriverStatus {
+  auto get_config_register = spi_protocol_->ReadRegister(reg::config::REG_ADDR);
+  if (get_config_register.first != types::DriverStatus::OK) {
+    return get_config_register.first;
+  }
+  utilities::Byte config_register(get_config_register.second);
+
+  if (encoding_scheme == CRCEncodingScheme::CRC_8BIT) {
+    config_register.ClearBit(reg::config::CRCO);
+  }
+
+  if (encoding_scheme == CRCEncodingScheme::CRC_16BIT) {
+    config_register.SetBit(reg::config::CRCO);
+  }
+
+  return spi_protocol_->WriteRegister(reg::config::REG_ADDR, config_register.Get());
+}
+
+auto NRF24L01Core::SetRFOutputPower(RFPowerSetting rf_power) noexcept -> types::DriverStatus {
+  auto get_rf_setup_reg = spi_protocol_->ReadRegister(reg::rf_setup::REG_ADDR);
+  if (get_rf_setup_reg.first != types::DriverStatus::OK) {
+    return get_rf_setup_reg.first;
+  }
+
+  auto rf_setup_reg = get_rf_setup_reg.second;
+
+  rf_setup_reg &= (0b11111001);
```

Comment:
Maybe give that mask a name that it is easier to identify what this mask means

Reply:
Done.